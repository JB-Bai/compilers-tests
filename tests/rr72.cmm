float;
struct ENrDq hsHowu5qa(){
	80907;
	if(sxx1Lqsiv) if(kfvlko3((YIMa2Se5v) = ! 22.948, 0xEadFf[cur8CfK(GQaW1).ypoP6 + - 066.73e1 || 383.767 / 7728.56e1[0323]]) || aynGK) 0xe28;
	else{
		if(KPA2hWx) if(- 8.007E+6 && oUcEcG7Oi && xbhoJ < - GmfOa2(857.12e-0, 052647 < cZ1Ewwf(- 0x5A62 + 43.32 / 0xAbf), md1a4q - ohjF2Ey - GXEOov_R - Eukzzu0(((! 0546[t_smbVNv[532.3633] && XeZ3NcFNUJ]) = 95.549e3[rJX4Xq = ! 13.06]), 012634)) / OhWLkYV && tX1sZyVfV) return E7pic(84.4E-1) - 31.89 * QPegm9a;
	}
	074;
}
struct Zv_mC Tpu5tjH2(int SfBna90rqe){
	float tLMPx83m = - - sMpfEATI[hs1oQT] && kny_9HD;
	struct jLAi8 B8Nvf = - S1XMUU * 38.822E0;
	struct wWNWUGyGJ IZlc7e[0xdC][0025] = ! - IPdPO8Ud[XpU6Jz4bIe] = q2XyrDQfVxk;
	int l8mrBOEJ9Z, FSftxr = 0x00 - RXfGuZF = 0306046.QfoT;
	struct{
	}
	SYhrki2 = 607.46E+7 >= kDqPmuca();
	return ! 947.43 / QXcmKOo.Vuo5uFVO;
	- ! 0x4BD7 / ItrQmv();
}
struct{
	int aACuqylQ;
	struct pebfY5Y fV5fWAP5r = 1622.51E-7;
	float hUQr30Mx = 00.15 + 0x3bEB1[20.5240e-6 * IGOO00RN6 / 9982.1E8 + - 0230 / 0xf52F - 19.9102 - cS3SsHWIpt + 9250 / 591.TjCD1OPM || 67.27].TlZVQy4;
}
dfEMfko2, F6aDFpD[2263];

