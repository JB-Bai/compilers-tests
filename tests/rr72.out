Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: float
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          StructSpecifier (2)
            STRUCT
            Tag (2)
              ID: ENrDq
        FunDec (2)
          ID: hsHowu5qa
          LP
          RP
        CompSt (2)
          LC
          StmtList (3)
            Stmt (3)
              Exp (3)
                INT: 80907
              SEMI
            StmtList (4)
              Stmt (4)
                IF
                LP
                Exp (4)
                  ID: sxx1Lqsiv
                RP
                Stmt (4)
                  IF
                  LP
                  Exp (4)
                    Exp (4)
                      ID: kfvlko3
                      LP
                      Args (4)
                        Exp (4)
                          Exp (4)
                            LP
                            Exp (4)
                              ID: YIMa2Se5v
                            RP
                          ASSIGNOP
                          Exp (4)
                            NOT
                            Exp (4)
                              FLOAT: 22.948000
                        COMMA
                        Args (4)
                          Exp (4)
                            Exp (4)
                              INT: 962047
                            LB
                            Exp (4)
                              Exp (4)
                                Exp (4)
                                  Exp (4)
                                    ID: cur8CfK
                                    LP
                                    Args (4)
                                      Exp (4)
                                        ID: GQaW1
                                    RP
                                  DOT
                                  ID: ypoP6
                                PLUS
                                Exp (4)
                                  MINUS
                                  Exp (4)
                                    FLOAT: 667.299988
                              OR
                              Exp (4)
                                Exp (4)
                                  FLOAT: 383.766998
                                DIV
                                Exp (4)
                                  Exp (4)
                                    FLOAT: 77285.601562
                                  LB
                                  Exp (4)
                                    INT: 211
                                  RB
                            RB
                      RP
                    OR
                    Exp (4)
                      ID: aynGK
                  RP
                  Stmt (4)
                    Exp (4)
                      INT: 3624
                    SEMI
                  ELSE
                  Stmt (5)
                    CompSt (5)
                      LC
                      StmtList (6)
                        Stmt (6)
                          IF
                          LP
                          Exp (6)
                            ID: KPA2hWx
                          RP
                          Stmt (6)
                            IF
                            LP
                            Exp (6)
                              Exp (6)
                                Exp (6)
                                  Exp (6)
                                    MINUS
                                    Exp (6)
                                      FLOAT: 8007000.000000
                                  AND
                                  Exp (6)
                                    ID: oUcEcG7Oi
                                AND
                                Exp (6)
                                  Exp (6)
                                    ID: xbhoJ
                                  RELOP
                                  Exp (6)
                                    Exp (6)
                                      MINUS
                                      Exp (6)
                                        ID: GmfOa2
                                        LP
                                        Args (6)
                                          Exp (6)
                                            FLOAT: 857.119995
                                          COMMA
                                          Args (6)
                                            Exp (6)
                                              Exp (6)
                                                INT: 21927
                                              RELOP
                                              Exp (6)
                                                ID: cZ1Ewwf
                                                LP
                                                Args (6)
                                                  Exp (6)
                                                    Exp (6)
                                                      MINUS
                                                      Exp (6)
                                                        INT: 23138
                                                    PLUS
                                                    Exp (6)
                                                      Exp (6)
                                                        FLOAT: 43.320000
                                                      DIV
                                                      Exp (6)
                                                        INT: 2751
                                                RP
                                            COMMA
                                            Args (6)
                                              Exp (6)
                                                Exp (6)
                                                  Exp (6)
                                                    Exp (6)
                                                      ID: md1a4q
                                                    MINUS
                                                    Exp (6)
                                                      ID: ohjF2Ey
                                                  MINUS
                                                  Exp (6)
                                                    ID: GXEOov_R
                                                MINUS
                                                Exp (6)
                                                  ID: Eukzzu0
                                                  LP
                                                  Args (6)
                                                    Exp (6)
                                                      LP
                                                      Exp (6)
                                                        Exp (6)
                                                          LP
                                                          Exp (6)
                                                            NOT
                                                            Exp (6)
                                                              Exp (6)
                                                                INT: 358
                                                              LB
                                                              Exp (6)
                                                                Exp (6)
                                                                  Exp (6)
                                                                    ID: t_smbVNv
                                                                  LB
                                                                  Exp (6)
                                                                    FLOAT: 532.363281
                                                                  RB
                                                                AND
                                                                Exp (6)
                                                                  ID: XeZ3NcFNUJ
                                                              RB
                                                          RP
                                                        ASSIGNOP
                                                        Exp (6)
                                                          Exp (6)
                                                            FLOAT: 95549.000000
                                                          LB
                                                          Exp (6)
                                                            Exp (6)
                                                              ID: rJX4Xq
                                                            ASSIGNOP
                                                            Exp (6)
                                                              NOT
                                                              Exp (6)
                                                                FLOAT: 13.060000
                                                          RB
                                                      RP
                                                    COMMA
                                                    Args (6)
                                                      Exp (6)
                                                        INT: 5532
                                                  RP
                                        RP
                                    DIV
                                    Exp (6)
                                      ID: OhWLkYV
                              AND
                              Exp (6)
                                ID: tX1sZyVfV
                            RP
                            Stmt (6)
                              RETURN
                              Exp (6)
                                Exp (6)
                                  ID: E7pic
                                  LP
                                  Args (6)
                                    Exp (6)
                                      FLOAT: 8.440000
                                  RP
                                MINUS
                                Exp (6)
                                  Exp (6)
                                    FLOAT: 31.889999
                                  STAR
                                  Exp (6)
                                    ID: QPegm9a
                              SEMI
                      RC
              StmtList (8)
                Stmt (8)
                  Exp (8)
                    INT: 60
                  SEMI
          RC
      ExtDefList (10)
        ExtDef (10)
          Specifier (10)
            StructSpecifier (10)
              STRUCT
              Tag (10)
                ID: Zv_mC
          FunDec (10)
            ID: Tpu5tjH2
            LP
            VarList (10)
              ParamDec (10)
                Specifier (10)
                  TYPE: int
                VarDec (10)
                  ID: SfBna90rqe
            RP
          CompSt (10)
            LC
            DefList (11)
              Def (11)
                Specifier (11)
                  TYPE: float
                DecList (11)
                  Dec (11)
                    VarDec (11)
                      ID: tLMPx83m
                    ASSIGNOP
                    Exp (11)
                      Exp (11)
                        MINUS
                        Exp (11)
                          MINUS
                          Exp (11)
                            Exp (11)
                              ID: sMpfEATI
                            LB
                            Exp (11)
                              ID: hs1oQT
                            RB
                      AND
                      Exp (11)
                        ID: kny_9HD
                SEMI
              DefList (12)
                Def (12)
                  Specifier (12)
                    StructSpecifier (12)
                      STRUCT
                      Tag (12)
                        ID: jLAi8
                  DecList (12)
                    Dec (12)
                      VarDec (12)
                        ID: B8Nvf
                      ASSIGNOP
                      Exp (12)
                        Exp (12)
                          MINUS
                          Exp (12)
                            ID: S1XMUU
                        STAR
                        Exp (12)
                          FLOAT: 38.821999
                  SEMI
                DefList (13)
                  Def (13)
                    Specifier (13)
                      StructSpecifier (13)
                        STRUCT
                        Tag (13)
                          ID: wWNWUGyGJ
                    DecList (13)
                      Dec (13)
                        VarDec (13)
                          VarDec (13)
                            VarDec (13)
                              ID: IZlc7e
                            LB
                            INT: 220
                            RB
                          LB
                          INT: 21
                          RB
                        ASSIGNOP
                        Exp (13)
                          Exp (13)
                            NOT
                            Exp (13)
                              MINUS
                              Exp (13)
                                Exp (13)
                                  ID: IPdPO8Ud
                                LB
                                Exp (13)
                                  ID: XpU6Jz4bIe
                                RB
                          ASSIGNOP
                          Exp (13)
                            ID: q2XyrDQfVxk
                    SEMI
                  DefList (14)
                    Def (14)
                      Specifier (14)
                        TYPE: int
                      DecList (14)
                        Dec (14)
                          VarDec (14)
                            ID: l8mrBOEJ9Z
                        COMMA
                        DecList (14)
                          Dec (14)
                            VarDec (14)
                              ID: FSftxr
                            ASSIGNOP
                            Exp (14)
                              Exp (14)
                                Exp (14)
                                  INT: 0
                                MINUS
                                Exp (14)
                                  ID: RXfGuZF
                              ASSIGNOP
                              Exp (14)
                                Exp (14)
                                  INT: 101414
                                DOT
                                ID: QfoT
                      SEMI
                    DefList (15)
                      Def (15)
                        Specifier (15)
                          StructSpecifier (15)
                            STRUCT
                            LC
                            RC
                        DecList (17)
                          Dec (17)
                            VarDec (17)
                              ID: SYhrki2
                            ASSIGNOP
                            Exp (17)
                              Exp (17)
                                FLOAT: 6074599936.000000
                              RELOP
                              Exp (17)
                                ID: kDqPmuca
                                LP
                                RP
                        SEMI
            StmtList (18)
              Stmt (18)
                RETURN
                Exp (18)
                  Exp (18)
                    NOT
                    Exp (18)
                      FLOAT: 947.429993
                  DIV
                  Exp (18)
                    Exp (18)
                      ID: QXcmKOo
                    DOT
                    ID: Vuo5uFVO
                SEMI
              StmtList (19)
                Stmt (19)
                  Exp (19)
                    Exp (19)
                      MINUS
                      Exp (19)
                        NOT
                        Exp (19)
                          INT: 19415
                    DIV
                    Exp (19)
                      ID: ItrQmv
                      LP
                      RP
                  SEMI
            RC
        ExtDefList (21)
          ExtDef (21)
            Specifier (21)
              StructSpecifier (21)
                STRUCT
                LC
                DefList (22)
                  Def (22)
                    Specifier (22)
                      TYPE: int
                    DecList (22)
                      Dec (22)
                        VarDec (22)
                          ID: aACuqylQ
                    SEMI
                  DefList (23)
                    Def (23)
                      Specifier (23)
                        StructSpecifier (23)
                          STRUCT
                          Tag (23)
                            ID: pebfY5Y
                      DecList (23)
                        Dec (23)
                          VarDec (23)
                            ID: fV5fWAP5r
                          ASSIGNOP
                          Exp (23)
                            FLOAT: 0.000162
                      SEMI
                    DefList (24)
                      Def (24)
                        Specifier (24)
                          TYPE: float
                        DecList (24)
                          Dec (24)
                            VarDec (24)
                              ID: hUQr30Mx
                            ASSIGNOP
                            Exp (24)
                              Exp (24)
                                FLOAT: 0.150000
                              PLUS
                              Exp (24)
                                Exp (24)
                                  Exp (24)
                                    INT: 245425
                                  LB
                                  Exp (24)
                                    Exp (24)
                                      Exp (24)
                                        Exp (24)
                                          Exp (24)
                                            Exp (24)
                                              Exp (24)
                                                Exp (24)
                                                  FLOAT: 0.000021
                                                STAR
                                                Exp (24)
                                                  ID: IGOO00RN6
                                              DIV
                                              Exp (24)
                                                FLOAT: 998210011136.000000
                                            PLUS
                                            Exp (24)
                                              Exp (24)
                                                MINUS
                                                Exp (24)
                                                  INT: 152
                                              DIV
                                              Exp (24)
                                                INT: 62767
                                          MINUS
                                          Exp (24)
                                            FLOAT: 19.910200
                                        MINUS
                                        Exp (24)
                                          ID: cS3SsHWIpt
                                      PLUS
                                      Exp (24)
                                        Exp (24)
                                          INT: 9250
                                        DIV
                                        Exp (24)
                                          Exp (24)
                                            INT: 591
                                          DOT
                                          ID: TjCD1OPM
                                    OR
                                    Exp (24)
                                      FLOAT: 67.269997
                                  RB
                                DOT
                                ID: TlZVQy4
                        SEMI
                RC
            ExtDecList (26)
              VarDec (26)
                ID: dfEMfko2
              COMMA
              ExtDecList (26)
                VarDec (26)
                  VarDec (26)
                    ID: F6aDFpD
                  LB
                  INT: 2263
                  RB
            SEMI
